### ISSUE TITLE
Improve Apex Autocomplete

### ISSUE THREAD

**Issue by ChuckJonas** [forcedotcom/salesforcedx-vscode#920]
### Summary

When Autocomplete is run on an SObject, it returns Global objects (EG `System`, `ApexPages`), which are not valid given the current instance context.   It also only returns a very limited set of fields.

### Steps To Reproduce:

1.  Create new class call testing
2.  Add the following code:

```java
public class Testing {
    public Testing() {
        Account acc = new Account();
        acc.
    }
}
```
3. Trigger a completion request on `acc.`

### Expected result

I would see the SObject Instance methods and all of it's fields (including custom fields)

### Actual result

I see the SObject instance methods, the global System namespaces & and very limited set of fields.

### Additional information
<img width="403" alt="testing_cls_ _realself" src="https://user-images.githubusercontent.com/5217568/44371077-543cf500-a49a-11e8-873c-ce27ba60ec7e.png">

**VS Code Version**:
1.26.0

**SFDX CLI Version**:
salesforcedx@pre-release

**OS and version**:
osx 10.12.6

**Comment by ChuckJonas** [comment]
Adding to the List...

I can't seem to get `static` variables and functions to auto-complete either

**Comment by Shababsoft** [comment]
I am not able to refresh sObject definitions.

```
Starting SFDX: Refresh SObject Definitions

19:10:56.481 sfdx sobject definitions refresh
Failure performing sObject describe Must pass a username and/or OAuth options when creating an AuthInfo instance.
19:11:00.567 sfdx sobject definitions refresh ended with error Failure performing sObject describe Must pass a username and/or OAuth options when creating an AuthInfo instance.

19:11:00.568 sfdx sobject definitions refresh ended with exit code 1
```

**Comment by lcampos** [comment]
Closing this since it was addressed on release `v47.5.0`
